App/CanvasAppliance.cpp

```diff
@@ -100,9 +100,9 @@ CanvasAppliance::CanvasAppliance(Canvas * extCanvas, QObject * parent)
     // set the startup project mode
     setProjectMode(extCanvas->modeInfo()->projectMode());
 
-    ui.undoButton->setShortcut(QKeySequence("Ctrl+Z"));
+    ui.undoButton->setToolTip(QString(tr("Undo (%1)")).arg(ui.undoButton->shortcut().toString()));
     connect(ui.undoButton, SIGNAL(clicked()), m_extCanvas, SLOT(undoSlot()));
-    ui.redoButton->setShortcut(QKeySequence("Ctrl+Shift+Z"));
+    ui.redoButton->setToolTip(QString(tr("Redo (%1)")).arg(ui.redoButton->shortcut().toString()));
```

Comment:
Minor nitpick: `tr()` should return a QString already. The extra `QString()` is likely not needed.

Reply:
Indeed. Still very much a WIP that I just revived from its years-long death, requiring self-review first ;)
Anyways this code is changing in the next upcoming commit ;)